### PR TITLE
Enhancement/author

### PR DESCRIPTION
**Pull request by iiinaD**

### CODE REVIEW

File: src/Chirp.Core/AuthorRepository.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+using Chirp.Core.DataModel;
+
+namespace Chirp.Core;
+
+public interface IAuthorRepository
+{
+    public Author? GetAuthor(string? authorName = null, string? authorEmail = null);
```
Comment (JonathanVil) [inline src/Chirp.Core/AuthorRepository.cs:7]: I believe this should be two signatures, as this poses the questions:
1. Can an author be retrieved by both name and email at the same time?
2. Can an author by retrieved without any parameters?

This signature allows for both of these.

File: src/Chirp.Core/AuthorRepository.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            query = query.Where(a => a.Name == authorName);
+        }
+
+        if (authorEmail != null)
+        {
+            query = query.Where(author => author.Email == authorEmail);
+        }
+        var list = query.ToList();
+        return (list.Count == 0) ? null : list.First();
```
Comment (JonathanVil) [inline src/Chirp.Core/AuthorRepository.cs:43]: Use [`FirstOrDefault`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.firstordefault?view=net-8.0) to achieve same functionality, but avoid loading potentially multiple results

File: src/Chirp.Razor/ChirpService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+using Chirp.Core;
+using Chirp.Core.DataModel;
+using Chirp.Core.DataTransferObjects;
+using Chirp.Razor.Exceptions;
+
+namespace Chirp.Razor;
+
+public record CheepViewModel(string Author, string Message, string Timestamp);
```
Comment (esser2002) [inline src/Chirp.Razor/ChirpService.cs:8]: Record 'CheepViewModel' is never used

File: src/Chirp.Razor/ChirpService.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    }
+
+    private static string UnixTimeStampToDateTimeString(double unixTimeStamp)
+    {
+        // Unix timestamp is seconds past epoch
+        var dateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+        dateTime = dateTime.AddSeconds(unixTimeStamp);
+        return dateTime.ToString("MM/dd/yy H:mm:ss");
+    }
```
Comment (esser2002) [inline src/Chirp.Razor/ChirpService.cs:68]: Should this be handled in the service?
(I am actually not sure)

File: src/Chirp.Razor/ChirpService.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            return null;
+        }
+
+        var dto = new AuthorDTO {Name = author.Name, Email = author.Email};
+        return dto;
+    }
+    
+    public void CreateAuthor(string authorName, string authorEmail)
+    {
+        if (GetAuthor(authorName, authorEmail) == null)
```
Comment (esser2002) [inline src/Chirp.Razor/ChirpService.cs:85]: Should this be != and not == ?

**Review by esser2002 — DISMISSED** [review]

**Review by JonathanVil — CHANGES_REQUESTED** [review]

File: src/Chirp.Core/AuthorRepository.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    /// <summary>
+    /// Get an author from the database by email
+    /// </summary>
+    /// <param name="authorEmail">The email of the author</param>
+    /// <returns>The requested author</returns>
+    public Author? GetAuthorByEmail(string authorEmail)
+    {
+        return GetAuthor(authorEmail: authorEmail);
```
Comment (JonathanVil) [inline src/Chirp.Core/AuthorRepository.cs:38]: Maybe call FirstOrDefault directly on the Authors DbSet here?

File: src/Chirp.Core/AuthorRepository.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    /// <summary>
+    /// Get an author from the database by name
+    /// </summary>
+    /// <param name="authorName">The name of the author</param>
+    /// <returns>The requested author</returns>
+    public Author? GetAuthorByName(string authorName)
+    {
+        return GetAuthor(authorName);
```
Comment (JonathanVil) [inline src/Chirp.Core/AuthorRepository.cs:28]: Maybe call FirstOrDefault directly on the Authors DbSet here?

**Review by JonathanVil — CHANGES_REQUESTED** [review]

**Review by JonathanVil — APPROVED** [review]

**Pull request merged**